LIGitHubPR.js

```diff
@@ -0,0 +1,432 @@
+// ==/UserScript==
+
+// icons.tabIcon : Icon made from <a href="http://www.onlinewebfonts.com/icon">Icon Fonts</a> is licensed by CC BY 3.0
+
+var itemCount = 0;
+var totalFilesCountReached = false;
+var checkFileTotalInterval = null;
+var isRunning = false;
+
+var icons = {
+    folder: '&#128447;',
+    file: '&#128459;',
+    tabIcon: '<svg height="16" version="1.1" width="14" class="octicon octicon-tree-file" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 1000 1000" enable-background="new 0 0 1000 1000" xml:space="preserve"><metadata> Svg Vector Icons : http://www.onlinewebfonts.com/icon </metadata><g><g transform="translate(0.000000,511.000000) scale(0.100000,-0.100000)"><path d="M905.9,4987.4c-42.6-15.5-96.9-48.4-124-73.6c-106.6-98.8-102.7-44.6-102.7-1459c0-1482.3-7.7-1402.8,153.1-1503.6l87.2-56.2h1149h1149V-672.4c0-2735.9-1.9-2656.4,87.2-2691.3c17.4-5.8,441.8-11.6,941.7-11.6H5155v-546.4c0-302.3,9.7-579.3,19.4-616.2c23.3-87.2,145.3-207.3,234.5-232.5c93-27.1,3567.1-25.2,3664,0c40.7,11.6,98.8,42.6,131.7,71.7c120.1,100.8,116.3,52.3,116.3,1271v1112.2l-44.6,65.9c-23.3,34.9-73.6,85.3-108.5,108.5l-65.9,44.6h-829.3h-827.4l-430.1,213.1L6585-1668.3l-604.5-5.8l-606.5-5.8l-69.7-54.2c-137.6-104.6-139.5-110.4-145.3-755.7l-5.8-575.5h-813.8h-811.9v1567.5V69.7l63.9,9.7c79.5,15.5,1461,15.5,1519.1,0c42.6-9.7,42.6-11.6,48.4-585.2l5.8-577.4l52.3-77.5c31-46.5,85.3-95,135.6-120.1c85.3-40.7,91.1-40.7,1891.1-40.7c1226.5,0,1821.3,5.8,1858.2,21.3c65.9,25.2,155,106.6,191.8,176.3c23.3,42.6,27.1,242.2,27.1,1149c0,1222.6,3.9,1180-131.7,1284.6l-67.8,52.3l-854.5,9.7l-854.5,9.7l-416.6,207.3l-416.6,209.3h-569.7c-542.5,0-575.4-1.9-656.8-40.7c-50.4-25.2-104.6-73.6-135.6-120.1l-52.3-77.5l-5.8-579.4l-5.8-577.4h-813.8h-811.9v746v744l548.3,5.8l550.3,5.8l69.7,54.3c147.3,110.4,139.5,48.4,139.5,1284.6V4348l-54.3,81.4c-29.1,44.6-87.2,98.8-129.8,120.1c-71.7,36.8-112.4,38.8-901,38.8h-825.4l-410.8,209.3l-410.8,207.3l-561.9,3.9C1101.6,5012.5,964,5008.7,905.9,4987.4z M2442.4,4491.3l397.2-203.5l842.9-9.7l842.9-9.7V3241.6V2214.7l-1767.1-5.8l-1769-3.9v1251.7v1249.8l529-5.8l527-5.8L2442.4,4491.3z M6937.6,1274.9l428.2-213.1h821.5h823.5V25.2v-1036.6l-1767.1,3.9l-1769,5.8l-3.9,1220.7c-1.9,672.3,0,1232.3,3.9,1245.9c5.8,17.4,127.9,23.2,521.2,23.2h511.5L6937.6,1274.9z M6937.6-2193.4l430.1-213.1h821.5h821.5v-1036.6v-1036.6l-1767.1,3.9l-1769,5.8l-3.9,1245.9l-3.9,1243.9h521.2h521.2L6937.6-2193.4z"/></g></g></svg>'
+}
+
+function collectFiles() {
+    var tabCounter = document.getElementById('files_tab_counter');
+
+    if (tabCounter.length == 0) {
+        return;
+    }
+
+    var expectedFilesCount = document.getElementById('files_tab_counter').textContent.trim();
+    var files = document.querySelectorAll('.js-diff-progressive-container .link-gray-dark[title]');
+    var input = [];
+
+    if (parseInt(expectedFilesCount, 10) == parseInt(files.length, 10)) {
+        totalFilesCountReached = true;
```

Comment:
Also maybe we should trigger the BuildUI when it is true for some kind of auto-refresh (I don't know if this comment is smart ... As BuildUI call collectFiles (infinite loop in progress) :) )

Reply:
As discussed before, we should refactor the whole processes.